internal/remotewrite/remote_write_url.go

```diff
@@ -72,13 +76,17 @@ func WithDataSourceName(dataSourceName string) URLOption {
 			return
 		}
 
-		if u.Values != nil {
-			u.Values.Add(prometheusServerQueryParam, dataSourceName)
-		} else {
-			u.Values = url.Values{
-				prometheusServerQueryParam: []string{dataSourceName},
-			}
+		u.Values.Add(prometheusServerQueryParam, dataSourceName)
+	}
+}
+
+func WithCollectorName(collectorName string) URLOption {
```

Comment:
I started that way but then realized that it could be extended without modifying the current implementation/tests. for the other cases. 
No strong opinions here on my side

Reply:
I would avoid adding features we do not use to reduce the complexity. However, if you think that we need it for tests we can keep it, but the only test using it is simply testing the functionality and not using it